code/bmpman/bmpman.cpp

```diff
@@ -2094,13 +2182,19 @@ void bm_page_in_stop() {
 		if ((bm_bitmaps[i].type != BM_TYPE_NONE) && (bm_bitmaps[i].type != BM_TYPE_RENDER_TARGET_DYNAMIC) && (bm_bitmaps[i].type != BM_TYPE_RENDER_TARGET_STATIC)) {
 			if (bm_bitmaps[i].preloaded) {
 				if (bm_preloading) {
+					if (bm_bitmaps[i].type == BM_TYPE_EFF && bm_bitmaps[i].info.ani.eff.in_subdir) {
+						set_temp_subdir_pathtype(bm_bitmaps[i].filename);
+					}
+
 					if (!gr_preload(bm_bitmaps[i].handle, (bm_bitmaps[i].preloaded == 2))) {
 						mprintf(("Out of VRAM.  Done preloading.\n"));
 						bm_preloading = 0;
 					}
 				} else {
 					bm_lock(bm_bitmaps[i].handle, (bm_bitmaps[i].used_flags == BMP_AABITMAP) ? 8 : 16, bm_bitmaps[i].used_flags);
-					bm_unlock(bm_bitmaps[i].handle);
+					if (bm_bitmaps[i].ref_count >= 1) {
```

Comment:
If I understand the code of `bm_lock` correctly then this expression should always be true. `bm_lock` increments the ref_count so it is always at least 1 (if the function is successful).